File: src/main/java/me/testaccount666/serversystem/placeholderapi/PlaceholderApiSupport.java
package me.testaccount666.serversystem.placeholderapi;

import me.testaccount666.serversystem.placeholderapi.executables.PlaceholderExpansionWrapper;

public class PlaceholderApiSupport {
    private static PlaceholderExpansionWrapper _Wrapper;

    private PlaceholderApiSupport() {
        throw new IllegalStateException("Utility class cannot be instantiated");
    }

    public static boolean isPlaceholderApiInstalled() {
        try {
            Class.forName("me.clip.placeholderapi.PlaceholderAPI");
            return true;
        } catch (Throwable throwable) {
            return false;
        }
    }

    public static void registerPlaceholders() {
        if (!isPlaceholderApiInstalled()) return;

        if (_Wrapper != null) unregisterPlaceholders();

        _Wrapper = new PlaceholderExpansionWrapper();
        _Wrapper.register();
    }

    public static void unregisterPlaceholders() {

        if (!isPlaceholderApiInstalled()) return;

        if (_Wrapper == null) return;

        _Wrapper.unregister();
        _Wrapper = null;

    }
}

File: src/main/kotlin/me/testaccount666/serversystem/placeholderapi/PlaceholderApiSupport.kt
package me.testaccount666.serversystem.placeholderapi

import me.testaccount666.serversystem.placeholderapi.executables.PlaceholderExpansionWrapper

class PlaceholderApiSupport private constructor() {
    init {
        throw IllegalStateException("Utility class cannot be instantiated")
    }

    companion object {
        private var _Wrapper: PlaceholderExpansionWrapper? = null

        val isPlaceholderApiInstalled: Boolean
            get() {
                try {
                    Class.forName("me.clip.placeholderapi.PlaceholderAPI")
                    return true
                } catch (_: Throwable) {
                    return false
                }
            }

        fun registerPlaceholders() {
            if (!isPlaceholderApiInstalled) return
            unregisterPlaceholders()

            _Wrapper = PlaceholderExpansionWrapper()
            _Wrapper?.register()
        }

        fun unregisterPlaceholders() {
            if (!isPlaceholderApiInstalled) return

            _Wrapper?.unregister()
            _Wrapper = null
        }
    }
}
